ui/dashboards/src/components/EmptyDashboard/EmptyDashboard.tsx

```diff
@@ -35,50 +37,90 @@ export interface EmptyDashboardProps {
   /**
    * Components that are placed below the title and description that include
    * actions for the user to take (e.g. buttons or links). If not specified,
-   * the default "add panel" and "add variable" buttons will be displayed. Set
-   * to `false` to disable the default buttons.
+   * the default buttons will be displayed. Set to `false` to disable the default
+   * buttons.
    */
-  actions?: boolean | React.ReactNode;
+  actions?: JSX.Element | boolean;
+
+  /**
+   * Handler for clicking the edit button when the dashboard is in "view" mode.
+   * Required when using the default empty state.
+   */
+  onEditButtonClick?: () => void;
 }
 
 const DEFAULT_TITLE = "Let's get started";
-const DEFAULT_DESCRIPTION = 'We currently support time series charts, gauge charts, stat charts and more!';
-const DEFAULT_ACTIONS = (
-  <>
-    <AddPanelButton variant="outlined" color="secondary" label="Add Panel" fullWidth={true} />
-    <EditVariablesButton variant="outlined" color="secondary" label="Add Variables" fullWidth={true} />
-  </>
-);
+
+const DEFAULT_DESCRIPTION = {
+  edit: 'We currently support time series charts, gauge charts, stat charts and more!',
+  view: 'This dashboard is currently empty. Get Started by clicking the edit button.',
+};
 
 // Constants from specifics in designs to make the default messaging look good.
 const CONTAINER_WIDTH = '450px';
 const PRIMARY_CONTENT_WIDTH = '289px';
 
+const COMMON_BUTTON_PROPS = {
+  variant: 'outlined',
+  color: 'secondary',
+} as const;
+
+type EmptyDashboardActionsProps = Pick<EmptyDashboardProps, 'actions' | 'onEditButtonClick'> & {
+  isEditMode: boolean;
+};
+
+const EmptyDashboardActions = ({ actions, isEditMode, onEditButtonClick }: EmptyDashboardActionsProps) => {
+  if (actions && typeof actions !== 'boolean') {
+    // Custom actions
+    return actions;
+  }
+
+  if (actions === false) {
+    // Disable default actions
+    return null;
+  }
+
+  if (isEditMode) {
+    // Default edit mode actions
+    return (
+      <>
+        <AddPanelButton variant="outlined" color="secondary" label="Add Panel" fullWidth={true} />
+        <EditVariablesButton variant="outlined" color="secondary" label="Add Variables" fullWidth={true} />
+      </>
+    );
+  }
+
+  if (onEditButtonClick) {
+    // Default view mode actions
+    return <EditDashboardButton {...COMMON_BUTTON_PROPS} label="Edit Dashboard" onClick={onEditButtonClick} />;
+  }
+
+  return null;
+};
+
 /**
  * Communicate that a dashboard is empty and prompt the user to get started.
  */
 export const EmptyDashboard = ({
   title = DEFAULT_TITLE,
-  description = DEFAULT_DESCRIPTION,
-  actions,
+  description,
   additionalText,
+  ...otherProps
```

Comment:
I wouldn't be opposed to spelling out the `props`, but not a big deal either way

Reply:
I think this is an artifact of the stuff I was trying with the discriminating union. I should be able to fix it now that I reverted that.